Issue 213 notification system

l3n0ire opened the pull request:
<!--  
Please include a summary of the addition/fix. 
-->
## Details

- User will be notified whenever a user comments, tags or likes their posts
  - no option to turn on/off notifications yet (should I add this?)
- When a user clicks on a notification, the post modal opens and the notification is marked as viewed
- For now notifications are deleted after they have been viewed
- Mark all as read will mark all notifications as read


Closes #213

JoelWiebe commented:
@l3n0ire I like the way you've designed it. We should be good with everyone getting system notifications, maybe later if we have email notifications, those can be toggled off in some user settings. Deleting notifications after they are read is good for now and will help keep the notifications pane clean. Great work.

JoelWiebe commented:
@l3n0ire Am I looking in the right place for notifications? Should notifications be displayed in the top nav bar?

<img width="287" alt="Screen Shot 2022-04-12 at 6 31 10 PM" src="https://user-images.githubusercontent.com/6416247/163065568-5cc9541a-c5a2-4375-ab5b-c17edccd32e3.png">

l3n0ire commented:
@JoelWiebe I added the post names and fixed the bugs you mentioned. If the postname is really long and cannot fit on one line by itself, then it will be shortened, otherwise words that can't fit on a line will just be pushed to the next line

Ammar-T commented on src/app/components/canvas/canvas.component.html:
```diff
@@ -19,6 +19,7 @@
         matTooltip="Adjust Settings">
         <mat-icon>settings</mat-icon>
       </button>
+      <app-notification-dropdown [props]="{user:user, board:board}"></app-notification-dropdown>
```
Small change but can we do something like `[user]="user" [board]="board"` instead?

Ammar-T commented on src/app/components/notification-dropdown/notification-dropdown.component.ts:
```diff
@@ -0,0 +1,100 @@
+  }
+
+  handleNotificationUpdate = (notification:Notification) =>{
+    console.log("ran")
```
```suggestion
```

Ammar-T commented on src/app/services/canvas.service.ts:
```diff
@@ -0,0 +1,68 @@
+      
+    }
+
+    async modifyTag(userId:string,postId: string, value: object,): Promise<any>{
```
```suggestion
    async modifyTag(postId: string, value: object,): Promise<any>{
```
We can remove this and get the current user from `auth.service.ts`.

Ammar-T commented:
Aside from those changes, good work!
